### DISCUSSION TITLE
Is MapLibre based on Leaflet?

### DISCUSSION THREAD

**Discussion by jayarjo** [maplibre/maplibre-gl-js#350] [Q&A] [answered]
### Question

I'm a bit confused about all those different mapping libs these days. So I will be grateful for some clarification. 

We are currently using Leaflet, which proved to be too raw to efficiently work with complex styles, hence the reason to migrate to Mapbox. Since from what we knew Mapbox was based on Leaflet (is it still based on Leaflet?). But then we encountered that weird licensing on current Mapbox versions. And then we found Maplibre! So the question is - is Maplibre based on Leaflet? I do not see any Leaflet mentioned in dependencies. Is it even slightly related? Is it full rewrite/alternative of what Leaflet was doing?

**Comment by HarelM** [comment]
The short answer, no.
Maplibre is a branch of mapbox from the time the license was open.
Leaflet is totally different in code and capabilities.
Leaflet is great for the very basic stuff, I used it too in the past, but needed more complicated things and moved to mapbox and then maplibre.
Openlayer is also great but mapbox/libre are the best renders for vector tiles.

**Comment by wipfli** [comment]
https://news.ycombinator.com/item?id=27605656

**Comment by kylebarron** [comment, marked as answer]
It's also confusing because years ago... Mapbox actually _was_ based on leaflet. The https://github.com/mapbox/mapbox.js project was a Mapbox extension on top of Leaflet. You can see that library was started way back in 2012.

Then along the way Mapbox decided Leaflet wasn't performant enough and created Mapbox GL JS from scratch. Instead of using the browser's Canvas API like Leaflet does, it uses WebGL for performance. Then Maplibre was born as a fork of Mapbox GL JS when the latter's license changed to not be openly licensed.
